src/proofnetwork/xjdf/data/xjdfdocument.cpp

```diff
@@ -0,0 +1,207 @@
+        d->jobId = arg;
+        emit jobIdChanged(arg);
+    }
+}
+
+void XJdfDocument::setJobPartId(const QString &arg)
+{
+    Q_D(XJdfDocument);
+    if (arg != d->jobPartId) {
+        d->jobPartId = arg;
+        emit jobPartIdChanged(arg);
+    }
+}
+
+void XJdfDocument::setAuditPool(const AuditPoolSP &arg)
+{
+    Q_D(XJdfDocument);
+    if (arg != d->auditPool) {
+        d->auditPool = arg;
+        emit auditPoolChanged(arg);
+    }
+}
+
+void XJdfDocument::setProductList(const ProductListSP &arg)
+{
+    Q_D(XJdfDocument);
+    if (arg != d->productList) {
+        d->productList = arg;
+        emit productListChanged(arg);
+    }
+}
+
+XJdfDocumentSP XJdfDocument::create()
+{
+    XJdfDocumentSP result(new XJdfDocument());
+    initSelfWeakPtr(result);
+    return result;
+}
+
+XJdfDocumentSP XJdfDocument::fromFile(const QString &filePath)
+{
+    QFile file(filePath);
+    if (!file.open(QFile::ReadOnly))
+        return XJdfDocumentSP();
+
+    QXmlStreamReader reader(&file);
+    return fromXJdf(reader);
+}
+
+XJdfDocumentSP XJdfDocument::fromXJdf(QXmlStreamReader &reader)
+{
+    XJdfDocumentSP document = create();
+    document->d_func()->document = document;
+
+    while (!reader.atEnd() && !reader.hasError()) {
+        if (reader.isStartElement()) {
+            if (reader.name() == QStringLiteral("XJDF")) {
+                document->readAttributesFromXJdf(reader);
+                auto attributes = reader.attributes();
+                document->setJobId(attributes.value(QStringLiteral("JobID")).toString());
+                if (attributes.hasAttribute(QStringLiteral("JobPartID")))
+                    document->setJobPartId(attributes.value(QStringLiteral("JobPartID")).toString());
+            } else if (reader.name() == QStringLiteral("ProductList")) {
+                document->setProductList(ProductList::fromXJdf(reader, document));
+            } else if (reader.name() == QStringLiteral("AuditPool")) {
+                document->setAuditPool(AuditPool::fromXJdf(reader, document));
+            } else {
+                document->fillFromXJdf(reader);
+            }
+        } else if (reader.isEndElement()) {
+            if (reader.name() == QStringLiteral("XJDF"))
+                break;
+        }
+        reader.readNext();
+    }
+    document->setFetched(true);
+
+    if (document->jobId().isEmpty() || !document->auditPool() || !document->productList()) {
+        qCWarning(proofNetworkXJdfDataLog) << "XJDF Document not created. XML is corrupted.";
+        return XJdfDocumentSP();
+    }
+
+    return document;
+}
+
+void XJdfDocument::toXJdf(QXmlStreamWriter &writer, bool) const
+{
+    writer.setAutoFormatting(true);
+    writer.writeStartDocument();
+    writer.writeStartElement(QStringLiteral("XJDF"));
+
+    writer.writeDefaultNamespace(QStringLiteral("http://www.CIP4.org/JDFSchema_2_0"));
+    writer.writeNamespace(QStringLiteral("https://www.opensoftdev.com/profit"), QStringLiteral("profit"));
```

Comment:
We shouldn't write this namespace all the times. We either need to be aware of its usage inside or set it manually. Otherwise we will write it to 3rdparty xjdfs that we are processing through JMS